bin/multiqc_to_custom_tsv.py

```diff
@@ -0,0 +1,114 @@
+#!/usr/bin/env python
+# copied from nf-core/viralrecon and adjusted
+
+import os
+import sys
+import errno
+import argparse
+import yaml
+
+
+def parse_args(args=None):
+    Description = 'Create custom spreadsheet for pertinent MultiQC bowtie 2 metrics generated by the nf-core/mag pipeline.'
+    Epilog = "Example usage: python multiqc_to_custom_tsv.py"
+    parser = argparse.ArgumentParser(description=Description, epilog=Epilog)
+    parser.add_argument('-md', '--multiqc_data_dir', type=str, dest="MULTIQC_DATA_DIR", default='multiqc_data', help="Full path to directory containing YAML files for each module, as generated by MultiQC. (default: 'multiqc_data').")
+    parser.add_argument('-se', '--single_end', dest="SINGLE_END", action='store_true', help="Specifies that the input is single-end reads.")
+    return parser.parse_args(args)
+
+
+def make_dir(path):
+    if not len(path) == 0:
+        try:
+            os.makedirs(path)
+        except OSError as exception:
+            if exception.errno != errno.EEXIST:
+                raise
+
+
+# Find key in dictionary created from YAML file recursively
+# From https://stackoverflow.com/a/37626981
+def find_tag(d, tag):
+    if tag in d:
+        yield d[tag]
+    for k,v in d.items():
+        if isinstance(v, dict):
+            for i in find_tag(v, tag):
+                yield i
+
+
+def yaml_fields_to_dict(YAMLFile,AppendDict={},FieldMappingList=[]):
+    with open(YAMLFile) as f:
+        yaml_dict = yaml.safe_load(f)
+        for k in yaml_dict.keys():
+            key = k
+            if key not in AppendDict:
+                AppendDict[key] = {}
+            if FieldMappingList != []:
+                for i,j in FieldMappingList:
+                    val = list(find_tag(yaml_dict[k], j[0]))
+                    if len(val) != 0:
+                        val = val[0]
+                        if len(j) == 2:
+                            val = list(find_tag(val, j[1]))[0]
+                        if i not in AppendDict[key]:
+                            AppendDict[key][i] = val
+                        else:
+                            print('WARNING: {} key already exists in dictionary so will be overwritten. YAML file {}.'.format(i,YAMLFile))
+            else:
+                AppendDict[key] = yaml_dict[k]
+    return AppendDict
+
+# customized
+def metrics_dict_to_file(FileFieldList,MultiQCDataDir,OutFile,se):
+    MetricsDict = {}
+    FieldList = []
+    for yamlFile,mappingList in FileFieldList:
+        yamlFile = os.path.join(MultiQCDataDir,yamlFile)
+        if os.path.exists(yamlFile):
+            MetricsDict = yaml_fields_to_dict(YAMLFile=yamlFile,AppendDict=MetricsDict,FieldMappingList=mappingList)
+            FieldList += [x[0] for x in mappingList]
+        else:
+            print('WARNING: File does not exist: {}'.format(yamlFile))
+
+    if MetricsDict != {}:
+        make_dir(os.path.dirname(OutFile))
+        fout = open(OutFile,'w')
```

Comment:
you should use `with` for all file handles, even in "w" mode

Reply:
Thanks, will change that